### ISSUE TITLE
 request to improve vanilla slideshow with more features missing: when having lot of images there is an issue .... pull request is blocked to post it there 

### ISSUE THREAD

**Issue by rapgithub** [DimitriMikadze/vanilla-slideshow#5]
vanilla-slideshow works fine and it is great slideshow, but i wonder if you can add some improvement actions functions in the slider to control these aspects, some ideas of functions are:

sliderpreload (function to run before the slider is initialized)
sliderloaded (function to run after the slider is initialized and loaded)
active (function to retrieve the current active slide)
slidepreload (function to run before the next slide is loaded)
slideloading (function to run while the next slide is loading)
slideloaded (function to run after the next slide is loaded)
slideloaderror (function to determine if the next slide has failed to load)

this way I can use these functions to control the slideshow behavior, 
and I retrieve:

errors, run specific function before during or after slider or slide is loaded and be able to implement an  imageloaded scripting to run to load images nicely if there are small or bigger etc...

I like the idea of using the imageloaded javascripting library like the one from desandro 

https://imagesloaded.desandro.com

it will be a great improvement!
thanks for hearing

**Comment by rapgithub** [comment]
also I forget to mention that it will be nice the if slider can have the function to set an initial number of images to load before showing the slider this helps a lot if I have 100 images or more that need to finish load in memory before showing the slider, like now I see in the html code that are all inside the div vanilla-slide and need to load all... it takes a long time to wait...        

<div class="vanilla-slide">

the slider behaves so slow when having more images so if I can set a number of images to load initially to a certain amount will help in case of massive amount of images in fullscreen and then when i click next slide it will load this next image at this moment when requested...and i have the chance to put a loader spinner in between to show present image while loading the next one...

maybe using a json external file for the slides to be included and not inside  the container <div class="vanilla-slide"> it will help a lot when loading large amount of images...

also set a certain amount of images to load initially keep the slider smooth and sleak working loading only the next slides when pressing next...

thanks

**Comment by DimiMikadze** [comment]
Hey @rapgithub thanks for your feedback and ideas, much appreciated!

Unfortunately now i don't have time to make a progress for this repo, maybe you want to be a contributor?

**Comment by rapgithub** [comment]
Yes, it will be good to improve the features to make it more suitable for many cases... I will see what i can do about it and yes need time too to dedicate to this... of I can do it sooner let you know... and big thanks for your interest too to make a progress for this repo.. 👍

**Comment by solutionweb-ch** [comment]
Hello,

Thanks a lot for this project. It's really light and simple, all good.

My only wish is to have what rapgithub asked too : 

> active (function to retrieve the current active slide)

I inserted a video tag and I want it to play only when its slide is showing. The autoplay attribute of the video tag starts the video even if the slideshow don't show the slide containing the video.

I can't do it with something like : `if ($('.vanilla-active').is(':visible')) document.querySelector('.vanilla-slide video').play();`

Thank a lot.

My code :
```
<div id="vanilla-slideshow-container">
	<div id="vanilla-slideshow">
		<div class="vanilla-slide" data-src="img/rect4-01.jpg"></div>
		<div class="vanilla-slide" data-src="">
			<video muted loop playsinline width="100%" height="100%" id="videoloop">
				<source src="img/video/video1.mp4" type="video/mp4">
			</video>
		</div>
		<div class="vanilla-slide" data-src="img/rect4-02.jpg"></div>
		<div class="vanilla-slide" data-src="img/rect4-03.jpg"></div>
		<div class="vanilla-slide" data-src="img/rect4-04.jpg"></div>
		<div class="vanilla-slide" data-src="img/rect4-05.jpg"></div>
	</div>
	<div id="vanilla-slideshow-previous"></div>
	<div id="vanilla-slideshow-next"></div>
</div>
```
